source/openqasm/openqasm3/parser.py

```diff
@@ -555,14 +555,28 @@ def visitExpressionTerminator(self, ctx: qasm3Parser.ExpressionTerminatorContext
             return Constant(const_name)
         if ctx.Integer():
             return IntegerLiteral(int(ctx.Integer().getText()))
+        if ctx.HexIntegerLiteral():
+            return IntegerLiteral(
+                int(ctx.HexIntegerLiteral().getText()[2:].replace("_", ""), base=16)
+            )
+        if ctx.DecIntegerLiteral():
+            return IntegerLiteral(int(ctx.DecIntegerLiteral().getText().replace("_", "")))
+        if ctx.OctIntegerLiteral():
+            return IntegerLiteral(
+                int(ctx.OctIntegerLiteral().getText()[2:].replace("_", ""), base=8)
+            )
+        if ctx.BinIntegerLiteral():
+            return IntegerLiteral(
+                int(ctx.BinIntegerLiteral().getText()[2:].replace("_", ""), base=2)
+            )
         if ctx.RealNumber():
             return RealLiteral(float(ctx.RealNumber().getText()))
         if ctx.booleanLiteral():
             return BooleanLiteral(True if ctx.booleanLiteral().getText() == "true" else False)
         if ctx.Identifier():
             return Identifier(ctx.Identifier().getText())
-        if ctx.StringLiteral():
-            return StringLiteral(ctx.StringLiteral().getText()[1:-1])
+        if ctx.BitStringLiteral():
+            return BitStringLiteral(ctx.BitStringLiteral().getText()[1:-1])
```

Comment:
**question**: Should we strip the `_` here?

Reply:
Good point, added in https://github.com/Qiskit/openqasm/pull/337/commits/dc51d0df2e4ddc61a9fa8de21df9f162bb9282c9